### PR TITLE
Homework 40 completed

### PR DESCRIPTION
**Pull request by Borisko130**

### CODE REVIEW

File: src/main/java/com/dev/cinema/model/ShoppingCart.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+@Entity
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    @OneToOne
+    private User user;
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/model/ShoppingCart.java:21]: we can use @MapsId annotation here

File: src/main/java/com/dev/cinema/model/Ticket.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "tickets")
+public class Ticket {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne(cascade = CascadeType.ALL)
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/model/Ticket.java:17]: Are you sure that @OneToOne is the best choice here? If you delete the ticket, what will happen to movieSession?

File: src/main/java/com/dev/cinema/model/Ticket.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    private MovieSession movieSession;
+    @OneToOne
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/model/Ticket.java:19]: Are you sure that @OnetoOne is the best choice here?

File: src/main/java/com/dev/cinema/service/ShoppingCartService.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import com.dev.cinema.model.MovieSession;
+import com.dev.cinema.model.ShoppingCart;
+import com.dev.cinema.model.User;
+
+public interface ShoppingCartService {
+    /**
+     * This method is responsible to add a Ticket to the ShoppingCart
+     * @param movieSession Contains the information required for a ticket
+     * @param user - user who wants to buy a ticket for a specific MovieSession
+     */
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/service/ShoppingCartService.java:12]: comments

File: src/main/java/com/dev/cinema/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
Comment (MaliukDaria) [inline src/main/java/com/dev/cinema/model/ShoppingCart.java:16]: If we use @MapsId annotation below, we won't need @GeneratedValue

File: src/main/java/com/dev/cinema/service/ShoppingCartService.java (hunk 4 above)
Comment (Borisko130) [reply src/main/java/com/dev/cinema/service/ShoppingCartService.java:12]: It is JavaDoc.

Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/service/ShoppingCartService.java:12]: Yes, it's javaDoc, but it's clear what this method does, plus it looks strange, when we have app with only one javaDoc for particular method, so let's remove it

File: src/main/java/com/dev/cinema/model/ShoppingCart.java (hunk 5 above)
Comment (RomanDubovskyi) [reply src/main/java/com/dev/cinema/model/ShoppingCart.java:16]: \+

**Comment by RomanDubovskyi** [comment]
@MaliukDaria left good comments, take care of them first

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**